src/main/java/org/opensearch/neuralsearch/sparse/codec/ClusteredPostingTermsWriter.java

```diff
@@ -84,13 +90,29 @@ public void setFieldAndMaxDoc(FieldInfo fieldInfo, int maxDoc) {
         float cluster_ratio = Float.parseFloat(fieldInfo.attributes().get(CLUSTER_RATIO_FIELD));
         int nPostings = Integer.parseInt(fieldInfo.attributes().get(N_POSTINGS_FIELD));
         float summaryPruneRatio = Float.parseFloat(fieldInfo.attributes().get(SUMMARY_PRUNE_RATIO_FIELD));
-        // TODO: attach a lucene reader in the following CacheGatedForwardIndexReader
+        BinaryDocValues binaryDocValues = null;
```

Comment:
it's not used outside of try block, put it inside and same for fmt and readState